File: packages/ios/FormbricksSDK/FormbricksSDK/Model/Javascript/JsFileUploadMessage.swift
struct JsFileUploadMessage: Codable {
    let event: EventType
    let uploadId: String
    let fileUploadParams: FileUploadParams
}

struct FileUploadParams: Codable {
    let file: File
    let params: Params
}

struct File: Codable {
    let name: String
    let type: String
    let base64: String
}

struct Params: Codable {
    let surveyId: String
}

File: packages/android/formbricksSDK/src/main/java/com/formbricks/formbrickssdk/model/javascript/JsFileUploadMessage.kt
package com.formbricks.formbrickssdk.model.javascript

import com.formbricks.formbrickssdk.model.response.EventType
import com.google.gson.Gson
import com.google.gson.annotations.SerializedName

data class JsFileUploadMessage(
    @SerializedName("event") val event: EventType,
    @SerializedName("uploadId") val uploadId: String,
    @SerializedName("fileUploadParams") val fileUploadParams: FileUploadParams
){
    companion object {
        fun from(string: String): JsFileUploadMessage {
            return Gson().fromJson(string, JsFileUploadMessage::class.java)
        }
    }
}

data class FileUploadParams(
    @SerializedName("file") val file: FileData,
    @SerializedName("params") val params: Params
)

data class FileData(
    @SerializedName("name") val name: String,
    @SerializedName("type") val type: String,
    @SerializedName("base64") val base64: String
)

data class Params(
    @SerializedName("surveyId") val surveyId: String
)